testdata/p4_16_samples/psa-persistent-bmv2.p4

```diff
@@ -0,0 +1,146 @@
+#include <core.p4>
+#include <bmv2/psa.p4>
+
+struct EMPTY { };
+
+typedef bit<48>  EthernetAddress;
+
+header ethernet_t {
+    EthernetAddress dstAddr;
+    EthernetAddress srcAddr;
+    bit<16>         etherType;
+}
+
+header output_data_t {
+    bit<32> word0;
+    bit<32> word1;
+    bit<32> word2;
+    bit<32> word3;
+}
+
+struct headers_t {
+    ethernet_t       ethernet;
+    output_data_t    output_data;
+}
+
+struct metadata_t {
+}
+
+parser MyIP(
+    packet_in pkt,
+    out headers_t hdr,
+    inout metadata_t user_meta,
+    in psa_ingress_parser_input_metadata_t istd,
+    in EMPTY resubmit_meta,
+    in EMPTY recirculate_meta)
+{
+    state start {
+        pkt.extract(hdr.ethernet);
+        pkt.extract(hdr.output_data);
+        transition accept;
+    }
+}
+
+parser MyEP(
+    packet_in pkt,
+    out headers_t hdr,
+    inout metadata_t user_meta,
+    in psa_egress_parser_input_metadata_t istd,
+    in EMPTY normal_meta,
+    in EMPTY clone_i2e_meta,
+    in EMPTY clone_e2e_meta)
+{
+    state start {
+        transition accept;
+    }
+}
+
+control MyIC(
+    inout headers_t hdr,
+    inout metadata_t user_meta,
+    in    psa_ingress_input_metadata_t istd,
+    inout psa_ingress_output_metadata_t ostd)
+{
+    @persistent bit<16> reg[6];
```

Comment:
How is this better than having an extern? So far, I see only downsides.
Also, what's the initial value of this? How people can reason about it?